File: sprint4/By lessons/lesson10code1/lesson10code1.cs
List<Tuple<int, int, int>> naive_solution(int A, List<int> x) {
    int n = x.Count;
    List<Tuple<int, int, int>> triples = new List<Tuple<int, int, int>>();
    for (int i1 = 0; i1 < n; ++i1) {
        for (int i2 = i1 + 1; i2 < n; ++i2) {
            for (int i3 = i2 + 1; i3 < n; ++i3) {
                if (x[i1] + x[i2] + x[i3] == A) {
                    triples.Add(Tuple.Create(x[i1], x[i2], x[i3]));
                }
            }
        }
    }
    return triples;
}

File: sprint4/By lessons/lesson10code1/lesson10code1.swift
func naiveSolution(_ A: Int, _ x: [Int]) -> [(Int, Int, Int)] {
    let n = x.count
    var triples: [(Int, Int, Int)] = []

    for i1 in 0..<n {
        for i2 in (i1 + 1)..<n {
            for i3 in (i2 + 1)..<n {
                if x[i1] + x[i2] + x[i3] == A {
                    triples.append((x[i1], x[i2], x[i3]))
                }
            }
        }
    }
    
    return triples
}